### PR TITLE
Feat/add external volume

### PR DESCRIPTION
**Pull request by danielchg**
In order to access an already existing PersistentVolumeClaim from a job in a drone step, we have added the add claim from the .drone.yml as below:
```
volumes:
  claim:
    name: <your-claim>
```

### CODE REVIEW

File: docker/Dockerfile.linux.amd64 [hunk 1]
```diff
@@ -1,14 +1,24 @@
-FROM alpine:3.6 as alpine
+FROM golang:1.14-alpine as builder
```
Comment (bradrydzewski) [inline docker/Dockerfile.linux.amd64:1]: we prefer our current approach to building the binary outside the dockerfile, and would kindly ask to revert this change.

File: engine/linter/testdata/volume_claim.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  - ls /shared
+  - sleep 500
+
+# once the pipeline finished, check the host machine
+# to see if exists /tmp/drone/test/greetings.txt
+
+volumes:
+- name: shared
+  claim:
+    claimName: received-data-claim
```
Comment (bradrydzewski) [inline engine/linter/testdata/volume_claim.yml:36]: we use snake case to name our attributes, however, in this case we can simplify this from `claimName` to just `name`

File: samples/workspace_default.yml [hunk 3]
```diff
@@ -12,6 +12,7 @@ steps:
   commands:
   - echo "hello" > greetings.txt
   - df -T /drone/src
+  - sleep 100000
```
Comment (bradrydzewski) [inline samples/workspace_default.yml:15]: I think this can be reverted

**Comment by bradrydzewski** [comment]
thanks!  a few minor comments, but otherwise looks good

File: engine/convert.go [hunk 4]
```diff
@@ -117,6 +117,19 @@ func toVolumes(spec *Spec) []v1.Volume {
 			volumes = append(volumes, volume)
 		}
 
+		if v.Claim != nil {
+			volume := v1.Volume{
+				Name: v.Claim.ID,
+				VolumeSource: v1.VolumeSource{
+					PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{
+						ClaimName: v.Claim.ClaimName,
+						ReadOnly:  true,
```
Comment (bradrydzewski) [inline engine/convert.go:126]: one last question -- should this be hard-coded to readonly or should we provide the option for read write?

Comment (danielchg) [reply engine/convert.go:126]: yeah, we can change this, for our use case is not needed but sounds like a useful configuration.
what about to add the the following to the .drone.yml 
```
volumes:
- name: shared
  claim:
    name: received-data-claim
    read_only: <true|false>
```

**Comment by Elgarni** [comment]
@danielchg Thanks for PR! it's indeed needed.

One thing to note here is, if the volume is going to be writable, then permissions need to be handled as well. 

Volumes in kubernetes are mounted with `root:root` ownership. If the container is running is non-root, it won't be able to write to it.

To solve this, we can utilize `Pod.spec.securityContext.fsGroup`, which will mount the volume with the specified group, and will run all containers with this supplementary group. From the [K8s API](https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/types.go#L2882)  
>   **fsGroup**: A special supplemental group that applies to all containers in a pod. Some
     volume types allow the Kubelet to change the ownership of that volume to be
     owned by the pod: 
       1. The owning GID will be the FSGroup 
       2. The setgid bit is set (new files created in the volume will be owned by FSGroup) 
       3. The permission bits are OR'd with rw-rw---- If unset, the Kubelet will not modify the ownership and permissions of any volume.

**Comment by danielchg** [comment]
@Elgarni 
You right about the permissions issue, I have tested this and when a Docker image has the USER <user> defined different than root, depending of your posix permissions within the filesystem created in the PVC you can or can not write there. But after digging on this I don't know if change the permissions of an already existing filesystem should be the best option.

 I just tested to use the PodSecurityContext to set the RunAsUser with UID 0 (run as root) in the converter.go, and with this all the process on the containers within this pod run as root user, not matter what is deffined in the Dockerfile of the image, hence these can write within the PVC. I think that may be good idea to add some kind of parameter within the .drone.yml to set the UID that has the proper permissions to write in the PVC instead of change the permissions on the filesystem, what do you guys think?

**Comment by Elgarni** [comment]
> I think that may be good idea to add some kind of parameter within the .drone.yml to set the UID that has the proper permissions to write in the PVC instead of change the permissions on the filesystem

Changing the UID might disallow writing on an existing filesystem layer in the image. 

I believe the primary usecase for mounting volumes is either share data between containers, or provide a cache that can be utilized by a container or more during the CI build. That being said, I think the safest approach here is set GID through the `fsGroup`, which is available at `Pod.spec.securityContext.fsGroup`.

This approach is highlighted in k8s docs: [set-the-security-context-for-a-pod](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#set-the-security-context-for-a-pod)

So the .yaml could have a field like this:
```yaml
...
volumes:
- name: shared
  claim:
    name: some-pvc
    mount_group: 2000
```

**Comment by danielchg** [comment]
I have been testing with hardcoded values within the `convert.go` (check it out [here)](https://gist.github.com/danielchg/7c32cdf0b272143046db86b9620b61a5) without any success.

I tested with Kuberentes 1.15 and 1.18 just to be sure that the issue is not because of any API version, and the behaviour is the same on both, when mount the PVC is with root privileges not with GUID 1000. 

If you alreay has tested `Pod.spec.securityContext.fsGroup` please help me to understand how it works.

File: engine/spec.go [hunk 5]
```diff
@@ -114,6 +115,14 @@ type (
 		FieldPath string `json:"field_path,omitempty"`
 	}
 
+	// VolumeClaim ...
+	VolumeClaim struct {
+		ID        string `json:"id,omitempty"`
+		Name      string `json:"name,omitempty"`
+		ClaimName string `json:"claimName,omitempty"`
+		ReadOnly  bool   `json:"readOnly,omitempty"`
```
Comment (bradrydzewski) [inline engine/spec.go:123]: snake case please :)

**Comment by Elgarni** [comment]
@danielchg can you please share the output `df -h` and `ls -la mount-path` when run inside the container

**Comment by Elgarni** [comment]
Also let's look into the populated pod spec by k8s api: `kubectl get pod POD_NAME -o yaml`

File: engine/spec.go (hunk 5 above)
Comment (danielchg) [reply engine/spec.go:123]: ups, I forgot to update this

**Comment by danielchg** [comment]
> @danielchg can you please share the output `df -h` and `ls -la mount-path` when run inside the container

here it is this information:

```bash
/drone/src $ whoami
test
/drone/src $ id
uid=1000(test) gid=1000(test) groups=1000(test)
/drone/src $ df |grep shared
overlay              490691512 315587368 150108664  68% /shared
/drone/src $ ls -la /shared
total 24
drwxr-xr-x    4 root     root          4096 Apr 23 16:36 .
drwxr-xr-x    1 root     root          4096 Apr 23 16:49 ..
drwxr-xr-x    5 root     root          4096 Apr 23 10:04 .minio.sys
-rw-r--r--    1 root     root             6 Apr 23 16:34 greetings.txt
-rw-r--r--    1 root     root            13 Apr 23 16:35 greetings2.txt
drwxr-xr-x    2 root     root          4096 Apr 23 16:36 test
/drone/src $ 

```
and here is the pod definition created from the drone runner:

```yaml
apiVersion: v1
kind: Pod
metadata:
  annotations:
    DRONE_BUILD_FINISHED: "1587660566"
    DRONE_BUILD_STATUS: success
    DRONE_STAGE_FINISHED: "1587660566"
    DRONE_STAGE_STATUS: success
    io.drone: "true"
    io.drone.build.number: "1"
    io.drone.created: "1587660566"
    io.drone.expires: "1587667766"
    io.drone.protected: "false"
    io.drone.repo.name: ""
    io.drone.repo.namespace: ""
    io.drone.repo.slug: ""
    io.drone.stage.name: default
    io.drone.stage.number: "1"
    io.drone.system.host: ""
    io.drone.system.proto: ""
    io.drone.system.version: ""
    io.drone.ttl: 1h0m0s
  creationTimestamp: "2020-04-23T16:49:26Z"
  labels:
    io.drone: "true"
    io.drone.build.event: push
    io.drone.build.number: "1"
    io.drone.name: drone-aibiiagvboflqqg15ptr
    io.drone.repo.name: ""
    io.drone.repo.namespace: ""
  name: drone-aibiiagvboflqqg15ptr
  namespace: toolkit
  resourceVersion: "34296"
  selfLink: /api/v1/namespaces/toolkit/pods/drone-aibiiagvboflqqg15ptr
  uid: a19d0889-1b47-4d41-8e2f-7f2d1f857173
spec:
  containers:
  - args:
    - echo "$DRONE_SCRIPT" | /bin/sh
    command:
    - /bin/sh
    - -c
    env:
    - name: DRONE_STAGE_OS
    - name: DRONE_WORKSPACE
      value: /drone/src
    - name: CI_COMMIT_AUTHOR_AVATAR
    - name: DRONE_SOURCE_BRANCH
    - name: DRONE_REPO_LINK
    - name: DRONE_GIT_HTTP_URL
    - name: GIT_AUTHOR_NAME
      value: drone
    - name: DRONE_COMMIT
    - name: CI_REPO
    - name: DRONE_SYSTEM_HOSTNAME
    - name: DRONE_REPO_OWNER
    - name: DRONE_SCRIPT
      value: "\n\nif [ ! -z \"${DRONE_NETRC_FILE}\" ]; then\n\techo $DRONE_NETRC_FILE
        > $HOME/.netrc\n\tchmod 600 $HOME/.netrc\nfi\n\nunset DRONE_SCRIPT\nunset
        DRONE_NETRC_MACHINE\nunset DRONE_NETRC_USERNAME\nunset DRONE_NETRC_PASSWORD\nunset
        DRONE_NETRC_FILE\n\nset -e\n\n\necho + \"pwd\"\npwd\n\necho + \"sleep 100\"\nsleep
        100\n\necho + \"echo \\\"hello\\\" > /shared/greetings.txt\"\necho \"hello\"
        > /shared/greetings.txt\n"
    - name: CI_COMMIT_AUTHOR_NAME
    - name: CI
      value: "true"
    - name: DRONE_STAGE_TYPE
    - name: GIT_COMMITTER_NAME
      value: drone
    - name: DRONE_STAGE_NAME
      value: default
    - name: DRONE_COMMIT_MESSAGE
    - name: DRONE_BUILD_PARENT
      value: "0"
    - name: DRONE_COMMIT_LINK
    - name: GIT_AUTHOR_EMAIL
      value: noreply@drone
    - name: DRONE_BUILD_ACTION
    - name: GIT_COMMITTER_EMAIL
      value: noreply@drone
    - name: DRONE_STAGE_FINISHED
      value: "1587660566"
    - name: DRONE_BUILD_CREATED
      value: "1587660566"
    - name: DRONE_REPO_NAMESPACE
    - name: DRONE_TARGET_BRANCH
    - name: DRONE_BRANCH
    - name: DRONE_STAGE_STARTED
      value: "1587660566"
    - name: DRONE_BUILD_LINK
      value: :////1
    - name: DRONE_REPO_SCM
    - name: DRONE_REPO
    - name: CI_BUILD_NUMBER
      value: "1"
    - name: DRONE_DEPLOY_ID
      value: "0"
    - name: CI_REPO_REMOTE
    - name: CI_PARENT_BUILD_NUMBER
      value: "0"
    - name: DRONE_SYSTEM_VERSION
    - name: DRONE_BUILD_FINISHED
      value: "1587660566"
    - name: CI_COMMIT_REF
    - name: DRONE
      value: "true"
    - name: CI_COMMIT_BRANCH
    - name: CI_BUILD_TARGET
    - name: CI_BUILD_CREATED
      value: "1587660566"
    - name: DRONE_BUILD_NUMBER
      value: "1"
    - name: CI_REPO_NAME
    - name: DRONE_COMMIT_AUTHOR_AVATAR
    - name: DRONE_STAGE_ARCH
    - name: CI_REMOTE_URL
    - name: DRONE_COMMIT_BEFORE
    - name: DRONE_SYSTEM_HOST
    - name: DRONE_COMMIT_REF
    - name: DRONE_SYSTEM_PROTO
    - name: CI_COMMIT_MESSAGE
    - name: CI_BUILD_FINISHED
      value: "0"
    - name: DRONE_REPO_BRANCH
    - name: CI_BUILD_STARTED
      value: "0"
    - name: DRONE_DEPLOY_TO
    - name: DRONE_REPO_VISIBILITY
    - name: CI_REPO_PRIVATE
      value: "false"
    - name: DRONE_DOCKER_VOLUME_ID
      value: drone-pi2ks035mg7dlxh1jaje
    - name: DRONE_COMMIT_BRANCH
    - name: DRONE_STAGE_NUMBER
      value: "1"
    - name: DRONE_STAGE_DEPENDS_ON
    - name: CI_REPO_LINK
    - name: CI_COMMIT_AUTHOR
    - name: DRONE_COMMIT_AUTHOR_NAME
    - name: DRONE_COMMIT_SHA
    - name: CI_BUILD_EVENT
      value: push
    - name: DRONE_STAGE_KIND
    - name: DRONE_STAGE_VARIANT
    - name: DRONE_REMOTE_URL
    - name: GIT_TERMINAL_PROMPT
      value: "0"
    - name: DRONE_REPO_PRIVATE
      value: "false"
    - name: DRONE_STAGE_MACHINE
    - name: DRONE_GIT_SSH_URL
    - name: DRONE_REPO_NAME
    - name: DRONE_BUILD_EVENT
      value: push
    - name: DRONE_BUILD_STATUS
      value: success
    - name: CI_BUILD_LINK
    - name: CI_COMMIT_AUTHOR_EMAIL
    - name: DRONE_COMMIT_AUTHOR_EMAIL
    - name: CI_COMMIT_SHA
    - name: DRONE_BUILD_STARTED
      value: "1587660566"
    - name: DRONE_STAGE_STATUS
      value: success
    - name: CI_BUILD_STATUS
    - name: DRONE_COMMIT_AUTHOR
    - name: DRONE_COMMIT_AFTER
    - name: KUBERNETES_NODE
      valueFrom:
        fieldRef:
          apiVersion: v1
          fieldPath: spec.nodeName
    image: docker.io/library/usertest:local
    imagePullPolicy: IfNotPresent
    name: drone-f674mdhl77hx9nhlfv9n
    resources: {}
    securityContext:
      privileged: false
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /shared
      name: drone-atu03zmn21iz4kd4li3g
    - mountPath: /drone/src
      name: drone-pi2ks035mg7dlxh1jaje
    - mountPath: /run/drone
      name: drone-6j92mpnjatimh2sigkdh
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: default-token-s5z2f
      readOnly: true
    workingDir: /drone/src
  - args:
    - echo "$DRONE_SCRIPT" | /bin/sh
    command:
    - /bin/sh
    - -c
    env:
    - name: DRONE_SYSTEM_HOSTNAME
    - name: DRONE_STAGE_KIND
    - name: DRONE
      value: "true"
    - name: DRONE_BUILD_NUMBER
      value: "1"
    - name: DRONE_REPO_LINK
    - name: DRONE_GIT_HTTP_URL
    - name: CI_COMMIT_MESSAGE
    - name: DRONE_BUILD_FINISHED
      value: "1587660566"
    - name: CI_BUILD_LINK
    - name: DRONE_BUILD_CREATED
      value: "1587660566"
    - name: DRONE_COMMIT
    - name: DRONE_STAGE_TYPE
    - name: DRONE_COMMIT_SHA
    - name: GIT_AUTHOR_NAME
      value: drone
    - name: DRONE_COMMIT_AUTHOR
    - name: CI_REPO
    - name: DRONE_REPO_VISIBILITY
    - name: DRONE_REPO
    - name: DRONE_BUILD_STARTED
      value: "1587660566"
    - name: CI_COMMIT_AUTHOR_NAME
    - name: DRONE_DOCKER_VOLUME_ID
      value: drone-pi2ks035mg7dlxh1jaje
    - name: CI_BUILD_STARTED
      value: "0"
    - name: DRONE_STAGE_VARIANT
    - name: CI_BUILD_STATUS
    - name: DRONE_GIT_SSH_URL
    - name: CI_REPO_LINK
    - name: DRONE_REPO_SCM
    - name: DRONE_COMMIT_AUTHOR_AVATAR
    - name: CI
      value: "true"
    - name: DRONE_STAGE_DEPENDS_ON
    - name: CI_BUILD_CREATED
      value: "1587660566"
    - name: DRONE_STAGE_NUMBER
      value: "1"
    - name: DRONE_SCRIPT
      value: "\n\nif [ ! -z \"${DRONE_NETRC_FILE}\" ]; then\n\techo $DRONE_NETRC_FILE
        > $HOME/.netrc\n\tchmod 600 $HOME/.netrc\nfi\n\nunset DRONE_SCRIPT\nunset
        DRONE_NETRC_MACHINE\nunset DRONE_NETRC_USERNAME\nunset DRONE_NETRC_PASSWORD\nunset
        DRONE_NETRC_FILE\n\nset -e\n\n\necho + \"pwd\"\npwd\n\necho + \"ls /shared\"\nls
        /shared\n\necho + \"cat /shared/greetings.txt\"\ncat /shared/greetings.txt\n"
    - name: DRONE_REPO_BRANCH
    - name: CI_COMMIT_AUTHOR_EMAIL
    - name: DRONE_REPO_PRIVATE
      value: "false"
    - name: CI_COMMIT_SHA
    - name: DRONE_DEPLOY_ID
      value: "0"
    - name: CI_REPO_PRIVATE
      value: "false"
    - name: CI_COMMIT_REF
    - name: DRONE_DEPLOY_TO
    - name: DRONE_SYSTEM_PROTO
    - name: DRONE_COMMIT_AFTER
    - name: DRONE_BUILD_LINK
      value: :////1
    - name: GIT_COMMITTER_NAME
      value: drone
    - name: CI_BUILD_EVENT
      value: push
    - name: CI_REMOTE_URL
    - name: DRONE_TARGET_BRANCH
    - name: CI_COMMIT_AUTHOR_AVATAR
    - name: CI_REPO_NAME
    - name: GIT_TERMINAL_PROMPT
      value: "0"
    - name: DRONE_COMMIT_BEFORE
    - name: DRONE_BUILD_ACTION
    - name: CI_PARENT_BUILD_NUMBER
      value: "0"
    - name: DRONE_STAGE_STARTED
      value: "1587660566"
    - name: CI_COMMIT_AUTHOR
    - name: DRONE_COMMIT_AUTHOR_EMAIL
    - name: GIT_AUTHOR_EMAIL
      value: noreply@drone
    - name: DRONE_BUILD_EVENT
      value: push
    - name: DRONE_COMMIT_LINK
    - name: CI_COMMIT_BRANCH
    - name: DRONE_BUILD_STATUS
      value: success
    - name: DRONE_STAGE_OS
    - name: DRONE_SYSTEM_VERSION
    - name: DRONE_REPO_NAMESPACE
    - name: DRONE_COMMIT_BRANCH
    - name: DRONE_STAGE_MACHINE
    - name: GIT_COMMITTER_EMAIL
      value: noreply@drone
    - name: CI_REPO_REMOTE
    - name: DRONE_BRANCH
    - name: DRONE_WORKSPACE
      value: /drone/src
    - name: DRONE_BUILD_PARENT
      value: "0"
    - name: DRONE_SOURCE_BRANCH
    - name: DRONE_SYSTEM_HOST
    - name: DRONE_STAGE_FINISHED
      value: "1587660566"
    - name: DRONE_REPO_NAME
    - name: DRONE_COMMIT_REF
    - name: CI_BUILD_TARGET
    - name: DRONE_STAGE_NAME
      value: default
    - name: DRONE_COMMIT_MESSAGE
    - name: DRONE_REMOTE_URL
    - name: DRONE_STAGE_ARCH
    - name: DRONE_COMMIT_AUTHOR_NAME
    - name: CI_BUILD_FINISHED
      value: "0"
    - name: CI_BUILD_NUMBER
      value: "1"
    - name: DRONE_STAGE_STATUS
      value: success
    - name: DRONE_REPO_OWNER
    - name: KUBERNETES_NODE
      valueFrom:
        fieldRef:
          apiVersion: v1
          fieldPath: spec.nodeName
    image: drone/placeholder:1
    imagePullPolicy: IfNotPresent
    name: drone-ng5ahtyprt4fybwdpu2y
    resources: {}
    securityContext:
      privileged: false
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /shared
      name: drone-atu03zmn21iz4kd4li3g
    - mountPath: /drone/src
      name: drone-pi2ks035mg7dlxh1jaje
    - mountPath: /run/drone
      name: drone-6j92mpnjatimh2sigkdh
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: default-token-s5z2f
      readOnly: true
    workingDir: /drone/src
  dnsConfig: {}
  dnsPolicy: ClusterFirst
  enableServiceLinks: true
  nodeName: toolkit
  priority: 0
  restartPolicy: Never
  schedulerName: default-scheduler
  securityContext:
    fsGroup: 1000
    runAsUser: 1000
  serviceAccount: default
  serviceAccountName: default
  terminationGracePeriodSeconds: 30
  tolerations:
  - effect: NoExecute
    key: node.kubernetes.io/not-ready
    operator: Exists
    tolerationSeconds: 300
  - effect: NoExecute
    key: node.kubernetes.io/unreachable
    operator: Exists
    tolerationSeconds: 300
  volumes:
  - emptyDir: {}
    name: drone-pi2ks035mg7dlxh1jaje
  - downwardAPI:
      defaultMode: 420
      items:
      - fieldRef:
          apiVersion: v1
          fieldPath: metadata.annotations
        path: env
    name: drone-6j92mpnjatimh2sigkdh
  - name: drone-atu03zmn21iz4kd4li3g
    persistentVolumeClaim:
      claimName: received-data-claim
  - name: default-token-s5z2f
    secret:
      defaultMode: 420
      secretName: default-token-s5z2f
status:
  conditions:
  - lastProbeTime: null
    lastTransitionTime: "2020-04-23T16:49:26Z"
    status: "True"
    type: Initialized
  - lastProbeTime: null
    lastTransitionTime: "2020-04-23T16:49:28Z"
    status: "True"
    type: Ready
  - lastProbeTime: null
    lastTransitionTime: "2020-04-23T16:49:28Z"
    status: "True"
    type: ContainersReady
  - lastProbeTime: null
    lastTransitionTime: "2020-04-23T16:49:26Z"
    status: "True"
    type: PodScheduled
  containerStatuses:
  - containerID: docker://a9b4b86eabde3a5c54c3a64621be6012b5bdc015da5606e6e7c10c1fbf298803
    image: usertest:local
    imageID: docker://sha256:9d524579f13b6a592a8157eaf03cf8e601166178887ec5630879353b81575dc8
    lastState:
      terminated:
        containerID: docker://2f3f45af98668896418b327f6eb53ec6f2a7e9b661fc593a94463650894f6f4c
        exitCode: 2
        finishedAt: "2020-04-23T16:49:28Z"
        reason: Error
        startedAt: "2020-04-23T16:49:27Z"
    name: drone-f674mdhl77hx9nhlfv9n
    ready: true
    restartCount: 1
    state:
      running:
        startedAt: "2020-04-23T16:49:28Z"
  - containerID: docker://bf5a48e4a3a92c1b4a84b881d7eef202ef4f4ef56b28e66fb0f03def087d0d96
    image: drone/placeholder:1
    imageID: docker-pullable://drone/placeholder@sha256:1b791b37e0589c7f1470fe0210ec0e7ef1e1de479c8f742d25942d2b4617bc07
    lastState: {}
    name: drone-ng5ahtyprt4fybwdpu2y
    ready: true
    restartCount: 0
    state:
      running:
        startedAt: "2020-04-23T16:49:27Z"
  hostIP: 172.17.0.2
  phase: Running
  podIP: 172.18.0.17
  qosClass: BestEffort
  startTime: "2020-04-23T16:49:26Z"

```

all these results are got  by adding the following content to `convert.go`

```go
func toPod(spec *Spec) *v1.Pod {
	var RunAsUID int64 = 1000
	var FSGuid int64 = 1000
	return &v1.Pod{
		ObjectMeta: metav1.ObjectMeta{
			Name:        spec.PodSpec.Name,
			Namespace:   spec.PodSpec.Namespace,
			Annotations: spec.PodSpec.Annotations,
			Labels:      spec.PodSpec.Labels,
		},
		Spec: v1.PodSpec{
			ServiceAccountName: spec.PodSpec.ServiceAccountName,
			SecurityContext: &v1.PodSecurityContext{
				RunAsUser: &RunAsUID,
				FSGroup:   &FSGuid,
			},
			RestartPolicy:    v1.RestartPolicyNever,
			Volumes:          toVolumes(spec),
			Containers:       toContainers(spec),
			NodeName:         spec.PodSpec.NodeName,
			NodeSelector:     spec.PodSpec.NodeSelector,
			Tolerations:      toTolerations(spec),
			ImagePullSecrets: toImagePullSecrets(spec),
			HostAliases:      toHostAliases(spec),
			DNSConfig:        toDnsConfig(spec),
		},
	}
}
```

**Comment by danielchg** [comment]
anyway I think that should be a good a idea to open a new issue in order manage the permissions issues adding more options to the `.drone.yml` not just with the volumes, maybe you would like to scale privileges in a job. With these changes I think is 100% usable, actually we are using it right now ;p

**Comment by Elgarni** [comment]
I see. After looking into [this issue](https://github.com/kubernetes/kubernetes/issues/2630), it turns out some volumes don't allow changing the ownership, hence `Kubelet` skips that when mounting. Are you using `Hostpath` for provisioning the pvc?

Anyways, like you said, this permissions/ownership can be handled in a diff PR discussion.

**Comment by danielchg** [comment]
yes, the local environment where I'm testing it I use hostPath, and after read ther issue that you shared now I undertand this behaviour, with hostPath is dangerous to update the permissions. Maybe in another provisioner like NFS, GlusterFS, EFS, ... this approach should works.

**Comment by Elgarni** [comment]
That makes sense. I believe since `fsGroup` is not special to PVCs, it should be handled at the volume declaration level. I can look into it sometime later. 

This PR LGTM

File: engine/spec.go (hunk 5 above)
Comment (danielchg) [reply engine/spec.go]: I also forgo to conver to snake case the readOnly label, now it's done.
